buildengine/engine.go

```diff
@@ -149,6 +151,28 @@ func (e *Engine) Import(ctx context.Context, schema *schema.Module) {
 
 // Build attempts to build the specified modules, or all local modules if none are provided.
 func (e *Engine) Build(ctx context.Context, modules ...string) error {
+	return e.buildWithCallback(ctx, nil, modules...)
+}
+
+// Deploy attempts to build and deploy the specified modules, or all local modules if none are provided.
+func (e *Engine) Deploy(ctx context.Context, replicas int32, waitForDeployOnline bool, modules ...string) error {
```

Comment:
I don't have a better idea really, but passing the same number of replicas for all deployments probably isn't what we'll want longer term.

Reply:
Yeah same. I wasn't 100% sure what to do here either, but open to improvements. Maybe we give this a go and if we have some use cases in the future that need something else we can update.